### PR TITLE
ft2-287 latest oc packages rc11

### PR DESCRIPTION
**Pull request by andycyberian**
Latest dev branch of OC source merged into SFA OC dev branch
New rc11 OC packages created and added to project
All project references updated including Fluid and NewtonSoft
All breaking changes addressed

Missing admin user recipe property #9092
Performance optimizations #9075
Correct TimeFieldIndex Time column type #9073
Update Fluid #9072
Ignore duplicate content item version ids when importing batches #9069
Fixes #9039 view buffer OutOfRangeException #9062
Upgrading to YesSql 3 #9059
Prevent some razor compilation stuff in production #9047
Fixes #8785 #8967 #9064 Use DeferredTask in ContentStep #9046
Fixes #9018 Shape AddAsync() in place of Add(). #9030
Remove enable rules feature from layers migration #9028
Fixes #9008 view buffer already rendered #9010
Introduce pooled view buffer #8999
Reduce some Liquid tags allocations #8998
Newtonsoft.Json 13.0.1 #8995
Fix OpenIdApplicationStep to include ResponseType rst: permissions #8994
Show DisplayMode and Editor on Parts in Content definition #8993
Remove antiforgery token in Taxonomy Tags view #8988
Workaround of the ViewBufferTextWriter issue #8984
Fixes #8979 Save Settings NRE #8981
Make ShellScope.StartAsyncFlow public #8978 
Missing Liquid ISite.BaseUrl accessor #8972 
Auto enabling Feature dependencies #8968
Remove unecessary casts from liquid filters #8966 
Add widget button opacity #8961 
Update OrchardCore.Lucene doc #8958 
Update SQL Indexing doc #8957 
Fixes #8948 Liquid ShapeTag #8956
Update OrchardCore.Alias doc #8954 
Update OrchardCore.Autoroute Doc #8953 
Update OrchardCore.Title Doc #8952 
Fix duplicate key in setup tenant workflow #8951 
Update GraphQL documentation #8947
Fix typo in DynamicCache liquid example for a cache Query #8946 
Try to fix docker windows push #8945 
Missing Liquid Site Properties #8944 
WF CommitTransaction in a separate WF Feature (Lombiq Technologies: OCORE-47) #8938 
Shortcode / Liquid Content conflict #8937
DetailAdmin Titles Tweaks  #8933 
Fixes #8914 Shortcode rendered once #8932 
Fixes Notifier rendering only one message #8931 
Code Mirror 5.60.0 #8930 
Font Awesome 5.15.3 #8928
Fixes #8915 Liquid Culture, Request, HttpContext Accessors #8926 
Add Liquid User Identity Accessor #8925 
Add Liquid User Identity Accessor #8925 
Fluid IHtmlContent Converter #8924 
Make radios and checkboxes more contrasted in dark mode #8909 
Fix ChildContent rendering in liquid #8908 
StringHtmlContent to HtmlContentString #8907 
Map Shape's public properties #8905 
Updating Github Actions to Node 15 #8901
Handle errors triggered by login event handlers #8899 
Cypress 6.8.0 #8897 
Fix OpenId Background Tasks - Upgrade YesSql #8896
HostString tweak also in the ApiController #8895 
PagerShapes: Use IShapeFactory and cleanup #8894 
Fix npm build #8892
Rename invalid package.json name for OrchardCore.AdminDashboard #8890 
Default VS Code builds to net5.0 #8883 
Forgot to remove unused var in Lucene AdminController #8882 
ZoneOnDemand tweaks #8880 
WF Max Recursion Depth (Lombiq Technologies: OCORE-48) #8878 
Update Fluid #8876
Upgrade Fluid to 2.0.0-beta-1010 #8875
Cypress 6.7.1 #8874 
Fixes LiquidUserAccessor #8872 
Fix IsSectionDefined #8871 
Fix layers content type rule to not include stereotypes #8868 
Remove dynamic usages from IShapeTableProvider #8867
Use new .AddAsync() in contentitem tag helper #8863 
Only one TFM by default under Visual Studio #8861 
Move OpenID collection initialization to Core Startup #8857
Sql Indexing documentation #8853
MailKit, MimeKit 2.11.0 #8852 
Expose BackgroundTask Events (Lombiq Technologies: OCORE-50) #8851 
YesSql 2.0.0-beta-1770 and Remove duplicate index #8850 
Fixing WidgetsList rendering #8849 
Darkmode should work with multi tenants in the same browser instance #8847 
Update resources README.md #8845 
Fix menu arrows direction in RTL #8843 
Fix arrows for collapsible panels in RTL languages #8841 
Placement for dashboard widgets #8839 
Remaining OpenId classes to OpenID Collection Table  ready#8837
Fluid 2.0.0-beta-1009 #8836 
Fix liquid parsing for Lucene Queries params  ready#8833 
Arrows for collapsible panels #8831 
Added missing namespace OrchardCore.Taxonomies.Helper to TaxonomyOrchardHelperExtensions.cs #8828 
Admin UI : content list style fixes #8827 
Maintain route data on contents admin list pager #8826
Fix UserQueryService name #8825
Content Parts documentation #8824 
VS Test SDK 16.9.1 #8821 
Add Indexes to ReduceIndex Tables too #8818 
Prevent Tests from failing on Sqlite file locked #8817 
Admin UI : fix filterable list-group borders #8816
Bump OpenIddict to 3.0.2  dependencies#8815
Add referer property on Liquid Request object #8810 
MiniProfiler on Admin as feature #8809 
RequestUrlHost tweaks #8804 
Custom user filters  ready #8800 
Fix links to Drivers in Themes documentation #8799
New project dialog #8796 
Fixes #8789 ProjectTemplates build #8792 
Remove unused .props file #8788
Migration steps convention #8786 
use AddHostedService #8782 
Tenants HostString tweaks #8779 
Fixes KeyNotFound in LayerFilter #8778 
Complete Module MultiTargeting #8777 
Fix migrations from RC2 to latest dev #8776
Adding Lucene Tests #8775 
Lucene Query syntax documentation #8774
Use sr-only on filters and bulk actions submit buttons #8770 
window functions #8764 
BagPart: add dropdown with max height #8759 
Fix gitignore for project template files #8758 
Added try-catch block around SuperUser lookup for recipe migrations. #8748 
Serilog 4.0.0 #8746 
Theme BuildMultiTargeting #8742

### CODE REVIEW

**Comment by lazcool** [comment]
Basically looks sound, but ideally it should tighten up the null handling in the sync code, e.g. use nullable types, rather than switching off warnings, handling nulls in the context of the checks that have already happened in the proceeding code & not treating empty string the same as null. As the sync code can be quite tricky to get 100%, it would also be preferred if there could be unit tests to cover any changes (ensuring the test cases reflect how OC actually presents the data).

The only other comment is that the .sln file should be updated to refer to the new package set.

File: DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Fields/TaxonomyFieldGraphSyncer.cs [hunk 1]
```diff
@@ -59,50 +59,51 @@ public async Task AddSyncComponents(JObject contentItemField, IGraphMergeContext
                 contentItemField, context.ContentItemVersion, context.ContentManager);
 
             JObject taxonomyPartContent = taxonomyContentItem!.Content[nameof(TaxonomyPart)];
-            string termContentType = taxonomyPartContent[TermContentType]!.Value<string>();
-
-            string termRelationshipType = TermRelationshipType(termContentType);
```
Comment (Gary-Griffiths-Methods) [inline DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Fields/TaxonomyFieldGraphSyncer.cs:64]: Does there need to be a null check on termContentType before it's used?

File: OrchardCore.AuditTrail/Services/AuditTrailManager.cs [hunk 2]
```diff
@@ -103,7 +104,10 @@ public async Task AddAuditTrailEventAsync<TAuditTrailEventProvider>(AuditTrailCo
             Filters filters = null,
             AuditTrailOrderBy orderBy = AuditTrailOrderBy.DateDescending)
         {
-            var session = _session.Store.CreateSession(System.Data.IsolationLevel.ReadUncommitted);
+
+            var session = _session.Store.CreateSession();
+            
+            //var session = _session.Store.CreateSession(System.Data.IsolationLevel.ReadUncommitted);
```
Comment (Gary-Griffiths-Methods) [inline OrchardCore.AuditTrail/Services/AuditTrailManager.cs:110]: Remove commented code?

**Review by Gary-Griffiths-Methods — COMMENTED** [review]
A couple of comments for feedback

**Review by Gary-Griffiths-Methods — APPROVED** [review]

**Pull request merged**